### ISSUE TITLE
Enabling parallelization of write_raw_bids

### ISSUE THREAD

**Issue by adam2392** [mne-tools/mne-bids#438]
#### Describe the problem
I was playing around w/ using `joblib` the other day to try to create my BIDs dataset using `write_raw_bids`, but I realized there's a concurrency issue, specifically around the participants and scans.tsv files. 

I am wondering if there is an easily implementable soln. to allow those files to be joblib safe? This is more of an inquiry, then a suggestion, as I have no idea how to implement, but wanted to get some thoughts.

#### Describe your solution
I looked up some possible solns. and since Python isn't thread safe, that's the main issue. So potentially if we could lock the file somehow, then this could be joblib able? Or possibly locking the functions that do writing to scans.tsv and participants files.

See:
- https://github.com/harlowja/fasteners
- https://arstechnica.com/civis/viewtopic.php?t=1234639

I think if the soln. is easy, only those files need to be locked, and everything else is concurrency safe, since it's a mapping between files.

**Comment by jasmainak** [comment]
hummm ... interesting

out of curiosity how long does it take for you to write with and without joblibs? how many subjects do you have?

i will defer to @agramfort 's opinion here

**Comment by adam2392** [comment]
> hummm ... interesting
> 
> out of curiosity how long does it take for you to write with and without joblibs? how many subjects do you have?

Well rn doing the write over a set of say... 100 EDF files (spread out over say... 60 subjects) all approx the same size (200-500 MB), and using 4 cores, I cut the time by a factor of 4. That is... until the job errors out due to concurrency of trying to write into scans.tsv and participants.tsv/.json files at the same time.

If I make a mistake in my pipeline, it's generally easier to just re-run the whole thing, since there isn't an API to update the BIDS files in place. So I haven't timed it, but I would say it's quite awhile w/o joblib compared to w/.

**Comment by agramfort** [comment]
I had the issue last week when converting cam-can to bids and it took me 4h.

my reaction is that it's not something I'll do every day so I can live with
the non-parallel
job. The other way would be to add parallel support in mne-bids and write
at the end
the shared file but it's low priority for me.

>

**Comment by Andesha** [comment]
Hi

Resurrecting this issue as it's something I am facing in my daily workflows - is there any steam on this or any recommended things I can do to avoid the issue?

**Comment by sappelhoff** [comment]
> just to confirm - if I update to the latest release I should be fine?

No, there is still work to be done in mne-bids. Dan was merely pointing at a related problem and how it was recently solved in mne-python. I agree with Dan, that the filelock library is promising for mne-bids, too.

Pull requests on this topic are welcome @Andesha
